internal/mempool/mempool.go

```diff
@@ -1618,16 +1618,28 @@ func (mp *TxPool) maybeAcceptTransaction(tx *dcrutil.Tx, isNew, rateLimit, allow
 
 	// Only allow TSpends that have a valid Expiry.
 	if isTreasuryEnabled && isTSpend {
-		// Ensure TSpend Expiry is exactly right. We don't use a window
-		// here since there should be enough blocks to cover the
-		// margin. The IsExpired check has been performed at this
-		// point.
-		expiry := standalone.CalculateTSpendExpiry(nextBlockHeight,
-			mp.cfg.ChainParams.TreasuryVoteInterval,
-			mp.cfg.ChainParams.TreasuryVoteIntervalMultiplier)
-		if expiry != msgTx.Expiry {
-			str := fmt.Sprintf("Invalid TSPEND Expiry must be %v "+
-				"got %v", expiry, msgTx.Expiry)
+		// Shorter variable names for relevant chain parameters.
+		tvi := mp.cfg.ChainParams.TreasuryVoteInterval
+		mul := mp.cfg.ChainParams.TreasuryVoteIntervalMultiplier
+
+		// Ensure the TSpend expiry isn't too much in the future,
+		// before its voting is supposed to start. We arbitrarily
+		// define as "too much in the future" as the vote starting
+		// greater than or equal to two full voting windows in the
+		// future.
+		voteStart, err := standalone.CalculateTSpendWindowStart(msgTx.Expiry,
+			tvi, mul)
+		if err != nil {
+			str := fmt.Sprintf("Invalid tspend expiry %d: %v ",
+				msgTx.Expiry, err)
+			return nil, txRuleError(ErrTSpendInvalidExpiry, str)
+		}
+		voteStartDistantFuture := int64(voteStart) > nextBlockHeight &&
+			int64(voteStart)-nextBlockHeight >= int64(2*tvi*mul)
+		if voteStartDistantFuture {
+			str := fmt.Sprintf("Tspend voting too much in the "+
+				"future: nbh %d expiry %d voteStart %d",
```

Comment:
'nbh' might be a bit cryptic for an error message

Reply:
Agreed.  However, I should also note that this is a problem with the vast majority of the errors in the treasury PR as a whole.  The trace statements are even worse in this regard.

Reply:
I saw the trace logs but those aren't messages most users would ever see or read.

Reply:
Changed in the following.

```diff
@@ -1638,12 +1637,15 @@ func (mp *TxPool) maybeAcceptTransaction(tx *dcrutil.Tx, isNew, rateLimit, allow
 				msgTx.Expiry, err)
 			return nil, txRuleError(ErrTSpendInvalidExpiry, str)
 		}
+		voteStartThresh := int64(2 * tvi * mul)
+		blocksToVoteStart := int64(voteStart) - nextBlockHeight
 		voteStartDistantFuture := int64(voteStart) > nextBlockHeight &&
-			int64(voteStart)-nextBlockHeight >= int64(2*tvi*mul)
+			blocksToVoteStart >= voteStartThresh
 		if voteStartDistantFuture {
-			str := fmt.Sprintf("Tspend voting too much in the "+
-				"future: nbh %d expiry %d voteStart %d",
-				nextBlockHeight, msgTx.Expiry, voteStart)
+			str := fmt.Sprintf("Tspend voting too far in the "+
+				"future: voting starts in %d blocks while the "+
+				"voting threshold is %d blocks",
+				blocksToVoteStart, voteStartThresh)
 			return nil, txRuleError(ErrTSpendInvalidExpiry, str)
 		}
```
https://github.com/decred/dcrd/pull/2350/commits/fb6fc2261c49f45039158d93b74f745b54794965#diff-199f9284707ff9893bd12190ce896c6aR1640-R1648